connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -105,8 +105,13 @@ public WorkerSourceTask(ConnectorTaskId id,
     }
 
     @Override
-    public void initialize(Map<String, String> config) {
-        this.taskConfig = config;
+    public void initialize(TaskConfig taskConfig) {
+        try {
+            this.taskConfig = taskConfig.originalsStrings();
```

Comment:
This looks weird, although I see `originalsStrings` can throw a `ClassCastException`. Is that actually possible here or is the `try`/`catch` not actually necessary?